src/java/org/apache/poi/ss/formula/FormulaParser.java

```diff
@@ -271,7 +271,17 @@ private RuntimeException expected(String s) {
 
     /** Recognize an Alpha Character */
     private static boolean IsAlpha(int c) {
-        return Character.isLetter(c) || c == '$' || c=='_';
+        return IsLetter(c) || c == '$' || c=='_';
+    }
+
+    /** Recognize a Letter Character */
+    private static boolean IsLetter(int c) {
+        return Character.isLetter(c) || c >= 256;
+    }
+
+    /** Recognize a Letter Character or a Decimal Digit */
+    private static boolean IsLetterOrDigit(int c) {
+        return Character.isLetterOrDigit(c) || c >= 256;
     }
```

Comment:
See unicode whitespace issue from `isLetter`

Would this work (avoids duplicating the rule `c>=256`)?

    return Character.isDigit(c) || FormulaParser.IsLetter(c);